pkg/node/node_address_linux.go

```diff
@@ -42,15 +44,33 @@ func firstGlobalV4Addr(intf string) (net.IP, error) {
 		return nil, err
 	}
 
+	wishIP := GetInternalIPv4()
```

Comment:
I'm not sure that using the global value is the safest choice. It means that this is sensitive to the order of operations before calling this function. It would be clearer if we passed in `preferredIPv4` and `preferredIPv6`. This might mean propagating that value up the call-tree a little bit more but it will be clear when in time the IP value was taken to be used here.